File: src/Services/Matches/Matches.API/Application/Commands/CreateMatchCommand.cs
using System.Collections.Generic;
using System.Text.Json;
using System.Text.Json.Serialization;
using MediatR;
using RacketReel.Services.Matches.API.Application.Dtos;

namespace RacketReel.Services.Matches.API.Application.Commands;

public class CreateMatchCommand : IRequest<MatchDto>
{
    public IEnumerable<string> Players { get; set; }
    public string ServingFirst { get; private set; }
    public int Sets { get; private set; }
    public string SetType { get; private set; }
    public string FinalSetType { get; private set; }

    [JsonConstructor]
    public CreateMatchCommand(IEnumerable<string> players, string servingFirst, int sets, string setType, string finalSetType)
    {
        Players = players;
        ServingFirst = servingFirst;
        Sets = sets;
        SetType = setType;
        FinalSetType = finalSetType;
    }

    public override string ToString()
    {
        return JsonSerializer.Serialize(this);
    }
}

File: src/racketreel_mobile/lib/matches/create_match_command.dart
import 'package:json_annotation/json_annotation.dart';
import 'set_type.dart';
part 'create_match_command.g.dart';

@JsonSerializable()
class CreateMatchCommand {
  List<String> players;
  String servingFirst;
  int sets;
  SetType setType;
  SetType finalSetType;

  CreateMatchCommand(
    this.players,
    this.servingFirst,
    this.sets,
    this.setType,
    this.finalSetType,
  );

  factory CreateMatchCommand.fromJson(Map<String, dynamic> json) =>
      _$CreateMatchCommandToJson(json);

  Map<String, dynamic> toJson() => _$CreateMatchCommandToJson(this);
}
